C/c4Document.cc

```diff
@@ -472,6 +491,90 @@ bool c4doc_save(C4Document *doc,
     return false;
 }
 
+bool c4doc_setExpiration(C4Database *db, C4Slice docId, uint64_t timestamp, C4Error *outError)
+{
+    C4Document *doc = c4doc_get(db, docId, true, outError);
+    if(doc == NULL) {
+        return false;
+    }
+    c4doc_free(doc);
+    
+    time_t now = time(NULL);
+    if(timestamp <= now) {
+        recordError(C4Domain, kC4ErrorInvalidParameter, outError);
+        return false;
+    }
+    
+    CollatableBuilder tsBuilder;
+    tsBuilder << timestamp;
+    slice tsSlice = tsBuilder.data();
+    
+    Transaction t(db);
+    KeyStore expiryKvs(db, "expiry");
+    KeyStoreWriter writer = t(expiryKvs);
+    
+    Document existingDoc = writer.get(docId);
+    if(existingDoc.exists()) {
+        // Previous entry found
+        if(existingDoc.body().compare(tsSlice) == 0) {
+            // No change
+            return true;
+        }
+        
+        // Remove old entry
+        Document existingTs = writer.get(existingDoc.body());
+        CollatableBuilder builder;
+        removeDocId(builder, existingTs.body(), docId);
+        writer.set(existingDoc.body(), builder.data());
+    }
+    
+    Document existingTs = writer.get(tsSlice);
+    slice docIDs;
+    if(existingTs.exists()) {
+        docIDs = existingTs.body();
+    }
+    
+    CollatableBuilder builder = docIDs.size == 0 ? CollatableBuilder() : CollatableBuilder(Collatable::withData(docIDs));
+    builder << docId;
+    writer.set(tsSlice, builder.data());
+    writer.set(docId, tsSlice);
+    
+    return true;
+}
+
+void c4doc_cancelExpiration(C4Database *db, C4Slice docId)
```

Comment:
This function can be merged into `c4doc_setExpiration`. Just define the time value UINT64_MAX to mean "never". Then skip the last part of setExpiration if the time is "never", and instead delete the docID entry.